File: app/src/main/java/com/zl/weilu/androidut/ui/MainActivity.java
package com.zl.weilu.androidut.ui;

import android.app.AlertDialog;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.CheckBox;
import android.widget.Toast;

import com.zl.weilu.androidut.R;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private CheckBox checkbox;
    private String lifecycle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        lifecycle = "onCreate";

        checkbox = (CheckBox) this.findViewById(R.id.checkbox);
    }

    public void jump(View view){

        startActivity(new Intent(this, LoginActivity.class));

//        startActivity(new Intent(this, LoginMVPActivity.class));

//        startActivity(new Intent(this, LoginDaggerActivity.class));
    }

    public void showToast(View view){
        Toast.makeText(this,"Hello UT!",Toast.LENGTH_LONG).show();
    }

    public void showDialog(View view){
        AlertDialog alertDialog = new AlertDialog.Builder(this)
                .setMessage("Hello UT！")
                .setTitle("提示")
                .create();
        alertDialog.show();
    }

    public void inverse(View view){
        checkbox.setChecked(!checkbox.isChecked());
    }

    public String getLifecycleState(){
        return lifecycle;
    }

    @Override
    protected void onStart() {
        super.onStart();
        lifecycle = "onStart";
    }

    @Override
    protected void onResume() {
        super.onResume();
        lifecycle = "onResume";
    }

    @Override
    protected void onPause() {
        super.onPause();
        lifecycle = "onPause";
    }

    @Override
    protected void onStop() {
        super.onStop();
        lifecycle = "onStop";
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        lifecycle = "onRestart";
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        lifecycle = "onDestroy";
    }
}

File: app/src/main/java/com/zl/weilu/androidut/ui/MainActivity.kt
package com.zl.weilu.androidut.ui

import android.app.AlertDialog
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Toast
import com.zl.weilu.androidut.R
import com.zl.weilu.androidut.mvp.ui.LoginMVPActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var lifecycleState: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        lifecycleState = "onCreate"
    }

    fun jump(view: View) {

//        startActivity(Intent(this, LoginActivity::class.java))

          startActivity(Intent(this, LoginMVPActivity::class.java))

//          startActivity(Intent(this, LoginDaggerActivity::class.java))
    }

    fun showToast(view: View) {
        Toast.makeText(this, "Hello UT!", Toast.LENGTH_LONG).show()
    }

    fun showDialog(view: View) {
        val alertDialog = AlertDialog.Builder(this)
                .setMessage("Hello UT！")
                .setTitle("提示")
                .create()
        alertDialog.show()
    }

    fun inverse(view: View) {
        checkbox.isChecked = !checkbox.isChecked
    }

    override fun onStart() {
        super.onStart()
        lifecycleState = "onStart"
    }

    override fun onResume() {
        super.onResume()
        lifecycleState = "onResume"
    }

    override fun onPause() {
        super.onPause()
        lifecycleState = "onPause"
    }

    override fun onStop() {
        super.onStop()
        lifecycleState = "onStop"
    }

    override fun onRestart() {
        super.onRestart()
        lifecycleState = "onRestart"
    }

    override fun onDestroy() {
        super.onDestroy()
        lifecycleState = "onDestroy"
    }
}
